src/super_gradients/training/transforms/processing.py

```diff
@@ -0,0 +1,186 @@
+
+    Note: This transformation assume that dimensions of input image is equal or less than `output_size`.
+
+    :param output_size: Output image size (rows, cols)
+    :param pad_value:   Padding value for image
+    """
+
+    def __init__(self, output_size: Tuple[int, int], pad_value: int):
+        self.output_size = output_size
+        self.pad_value = pad_value
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, DetectionPadToSizeMetadata]:
+        shift_h, shift_w, pad_h, pad_w = _get_center_padding_params(input_size=image.shape, output_size=self.output_size)
+        processed_image = _shift_image(image, pad_h, pad_w, self.pad_value)
+
+        return processed_image, DetectionPadToSizeMetadata(shift_h=shift_h, shift_w=shift_w)
+
+    def postprocess_predictions(self, predictions: np.ndarray, metadata: DetectionPadToSizeMetadata) -> np.ndarray:
+        return _shift_bboxes(targets=predictions, shift_w=-metadata.shift_w, shift_h=-metadata.shift_h)
+
+
+class DetectionSidePadding(Processing):
+    """Preprocessing transform to pad image and bboxes to `output_size` shape (rows, cols).
+    Side padding, so that input image with bboxes will located on the side. Bboxes won't be affected.
+
+    Note: This transformation assume that dimensions of input image is equal or less than `output_size`.
+
+    :param output_size: Output image size (rows, cols)
+    :param pad_value:   Padding value for image
+    """
+
+    def __init__(self, output_size: Tuple[int, int], pad_value: int):
+        self.output_size = output_size
+        self.pad_value = pad_value
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, None]:
+        processed_image = _pad_image_on_side(image, output_size=self.output_size, pad_val=self.pad_value)
+        return processed_image, None
+
+    def postprocess_predictions(self, predictions: np.ndarray, metadata: None) -> np.ndarray:
+        return predictions
+
+
+class _Rescale(Processing, ABC):
+    """Resize image to given image dimensions WITHOUT preserving aspect ratio.
+
+    :param output_shape: (rows, cols)
+    """
+
+    def __init__(self, output_shape: Tuple[int, int], keep_aspect_ratio: bool):
+        self.output_shape = output_shape
+        self.keep_aspect_ratio = keep_aspect_ratio
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, RescaleMetadata]:
+        rescale_shape = self.output_shape
+        sy, sx = rescale_shape[0] / image.shape[0], rescale_shape[1] / image.shape[1]
+
+        if self.keep_aspect_ratio:
+            scale_factor = min(sy, sx)
+            sy, sx = (scale_factor, scale_factor)
+            rescale_shape = (int(image.shape[0] * sx), int(image.shape[1] * sy))
+
+        rescaled_image = _rescale_image(image, target_shape=rescale_shape)
+
+        return rescaled_image, RescaleMetadata(original_size=image.shape[:2], sy=sy, sx=sx)
+
+
+class DetectionRescale(_Rescale):
```

Comment:
I think we also need LongestMaxSizeRescale than preserves aspect ratio and resize image that it's longest size is of desired size.